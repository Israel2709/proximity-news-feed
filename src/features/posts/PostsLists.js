import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { selectAllPosts, fetchPosts } from './postsSlice'

import { Link } from 'react-router-dom'
import { Spinner } from '../../app/Components/Spinner'
import { PostCard } from '../../app/Components/PostCard'

export const PostsList = () => {
    const dispatch = useDispatch()
    const posts = useSelector( selectAllPosts )
    const postStatus = useSelector( state => state.posts.status )
    const error = useSelector( state => state.posts.error)

    let content

    if ( postStatus === 'loading' ) {
        content = <Spinner />
    } else if ( postStatus === 'succeeded' ) {
        content = posts.map( post => (
            <PostCard postData = { post } />
        ))
    } else if (postStatus === 'failed') {
        content = <div>{error}</div>
    }

    useEffect(() => {
        if ( postStatus === 'idle' ) {
            dispatch( fetchPosts() )
        }
    }, [postStatus, dispatch] )

    return(
        <div>
            { content }
        </div>
        
    )
}
