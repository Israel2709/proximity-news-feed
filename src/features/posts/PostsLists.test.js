import React from 'react'
import renderer from 'react-test-renderer'
import { store } from '../../app/store';
import { Provider } from 'react-redux';

import { PostsList} from './PostsLists'

it("Renders when there are posts to show", () => {
    const tree = renderer.create(<Provider store={store}><PostsList /></Provider>).toJSON()
    expect(tree).toMatchSnapshot();
})