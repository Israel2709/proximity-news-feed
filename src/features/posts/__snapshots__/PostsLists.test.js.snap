// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders when there are posts to show 1`] = `<div />`;
